doc/source/selections.rst

```diff
@@ -0,0 +1,440 @@
+    allowed.
+
+cyzone *externalRadius* *zMax* *zMin* *selection*
+    selects all atoms within a cylindric zone centered in the center of
+    geometry (COG) of a given selection, e.g. ``cyzone 15 4 -8 protein and
+    resid 42`` selects the center of geometry of protein and resid 42, and
+    creates a cylinder of external radius 15 centered on the COG. In z, the
+    cylinder extends from 4 above the COG to 8 below. Positive values for
+    *zMin*, or negative ones for *zMax*, are allowed.
+
+    .. versionchanged:: 0.10.0
+       keywords *cyzone* and *cylayer* now take *zMax* and *zMin* to be
+       relative to the COG of *selection*, instead of absolute z-values
+       in the box.
+
+point *x* *y* *z*  *distance*
+    selects all atoms within a cutoff of a point in space, make sure
+    coordinate is separated by spaces, e.g. ``point 5.0 5.0 5.0 3.5``
+    selects all atoms within 3.5 Angstroms of the coordinate (5.0, 5.0,
+    5.0)
+
+prop [abs] *property*  *operator*  *value*
+    selects atoms based on position, using *property* **x**, **y**, or
+    **z** coordinate. Supports the **abs** keyword (for absolute value) and
+    the following *operators*: **<, >, <=, >=, ==, !=**. For example,
+    ``prop z >= 5.0`` selects all atoms with z coordinate greater than 5.0;
+    ``prop abs z <= 5.0`` selects all atoms within -5.0 <= z <= 5.0.
+
+From version 0.6 onwards, some geometric selections (around, sphlayer,
+sphzone, point) can use a k-d tree based, fast search algorithm (about three
+times faster than the previous version). However, it does not take periodicity
+into account. The fast algorithm is the default for *around*. Periodicity is
+only taken into account with the
+:func:`~MDAnalysis.lib.distances.distance_array` functions via a minimum
+image convention (and this only works for rectangular simulation cells). If
+periodic boundary conditions should be taken into account then change the
+default behaviour of MDAnalysis by setting these two flags::
+
+  MDAnalysis.core.flags['use_periodic_selections'] = True
+  MDAnalysis.core.flags['use_KDTree_routines'] = False
+
+
+Similarity and connectivity
+---------------------------
+
+same *subkeyword* as *selection*
+    selects all atoms that have the same *subkeyword* value as any atom in
+    *selection*. Allowed *subkeyword* values are the atom properties: ``name,
+    type, resname, resid, segid, mass, charge, radius, bfactor, resnum``, the
+    groups an atom belong to: ``residue, segment, fragment``, and the atom
+    coordinates ``x, y, z``.
+
+byres *selection*
+    selects all atoms that are in the same segment and residue as selection,
+    e.g. specify the subselection after the byres keyword.  ``byres`` is a
+    shortcut to ``same residue as``
+
+bonded *selection*
+    selects all atoms that are bonded to selection
+    eg: ``select name H and bonded name O`` selects only hydrogens bonded to
+    oxygens
+
+Index
+-----
+
+bynum *index-range*
```

Comment:
We need to rewrite this and not call it "index" but "atom number" – anything different from "index". In reality it just adds 1 to the 0-based index. It would be nice if we were to use it for "atom numbers" obtained from the topology file, such as PDB AtomID or TPR index https://github.com/MDAnalysis/mdanalysis/issues/2364